dhis-2/dhis-web/dhis-web-dataentry/src/main/java/org/hisp/dhis/de/action/GetDataValuesForDataSetAction.java

```diff
@@ -335,8 +346,10 @@ public String execute()
                 lastUpdatedBy = registration.getLastUpdatedBy();
             }
 
-            locked = dataSetService.isLocked( currentUser, dataSet, period, organisationUnit, attributeOptionCombo,
-                null );
+            locked = dataSetService.isLocked( currentUser, dataSet, period, organisationUnit, null );
+            approved = dataApprovalService.isApproved( dataSet.getWorkflow(), period, organisationUnit,
```

Comment:
Many thanks for the nice contribution. One issue here is that the check for `dataApprovalService.isApproved` is also currently done within `dataSetService.isLocked`. This means we are doing the check twice, which is not great from a performance perspective, nor from a readability perspective.

Instead I suggest that we change the return type of dataSetService.isLocked to an enum, and rename the method to something like `getLockStatus`. The enum could have values such as OPEN, LOCKED, APPROVED, etc. That way the semantics is cleaner and performance is better.

Reply:
@larshelge Actually, this overload of `dataSetService.isLocked` doesn't call `dataApprovalService.isApproved`. It's called by the longer overload of `isLocked` that also calls `isApproved`.

I agree that the enum would be less confusing all around and could also be used for SMS.